src/aws/auth.rs

```diff
@@ -138,8 +152,21 @@ impl AwsAuthentication {
                 secret_access_key.inner(),
                 None,
             ))),
-            AwsAuthentication::File { .. } => {
-                Err("Overriding the credentials file is not supported.".into())
+            AwsAuthentication::File {
+                credentials_file,
+                profile,
+            } => {
+                // The SDK uses the default profile out of the box, but doesn't provide an optional
+                // type in the builder. We can just hardcode it so that everything works.
+                let profile_name = profile;
+                let profile_files = ProfileFiles::builder()
+                    .with_file(ProfileFileKind::Credentials, credentials_file)
+                    .build();
+                let profile_provider = ProfileFileCredentialsProvider::builder()
+                    .profile_files(profile_files)
+                    .profile_name(profile_name)
```

Comment:
```suggestion
                    .profile_name(profile)
```
🤔 shouldn't this work and avoid the `let` on line 161?

Reply:
Yeah, that should work now that it's not optional. I can go ahead and fix it.

Reply:
Nice, sorry for the late request on that 😅 I'll hit the merge button as soon as that's updated. Thanks for putting up with my endless requests 😄